app/models/annotation/handler/AnnotationInformationHandler.scala

```diff
@@ -29,25 +31,26 @@ trait AnnotationInformationHandler extends FoxImplicits {
 
   def provideAnnotation(identifier: ObjectId, user: Option[User])(implicit ctx: DBAccessContext): Fox[Annotation]
 
+  @silent // suppress warning about unused implicit ctx, as it is used in subclasses
   def nameForAnnotation(t: Annotation)(implicit ctx: DBAccessContext): Fox[String] =
     Fox.successful(t.id)
 
   def restrictionsFor(identifier: ObjectId)(implicit ctx: DBAccessContext): Fox[AnnotationRestrictions]
 
   def assertAllOnSameDataset(annotations: List[Annotation]): Fox[Boolean] = {
+    @tailrec
```

Comment:
Do we want to add this annotation to all tailrecursive functions? There are also other places, where we use tailrecursive functions.

Reply:
Yes, eventually I would like to see all tail recursive functions annotated. However, they are not super easy to find.
Actually this is part of a somewhat larger campaign to address the code style warnings produced by IntelliJ. This started from my trying to get rid of unnecessarily public members of any classes, and developed into a desire to get rid of all of these warnings. This is also related to the removing of superfluous enclosing blocks and unused vales. However, there is too much code to do all of that manually all at once. So this is why I clean a file here and there by and by. Same goes for the ${id.id} thing and single-line yield statements, let’s defer that to a later PR (or multiple ones to keep conflicts small). This does not have a particularly high priority. Do you think this approach is fair?